### DISCUSSION TITLE
Is it possible to import an existing AWS Account?

### DISCUSSION THREAD

**Discussion by umm0n** [gruntwork-io/knowledge-base#332] [Help] [answered]
Is it possible to import an existing AWS account, specifically an organization management account? Our org management account is outside of our IAC and we're looking into bringing it in.

[r:terragrunt](https://github.com/gruntwork-io/terragrunt)

**Comment by autero1** [comment, marked as answer]
It is possible, but successfully importing existing accounts highly depends on multiple factors: 

- How are your resources named and will you be able to retrofit the existing name to the modules? Certain AWS resources force re-creation if you attempt to change
- Are you currently using features not yet supported by Gruntwork modules?

I've done a few account "codifications" with varying success. With some accounts we've had almost 100% success, but with some we've settled for partial codification. Overall, the process is just

- Map out your resources and dependencies
- Create IaC configurations for the above
- Import the resources with `terragrunt import` (or `terraform import`). The Terraform documentation has good documentation on how to import existing resources. You can also import resources within modules and even resources within deeper module hierarchy, e.g. `terraform import module.mymodule.module.nested_module.aws_msk_cluster.msk arn:aws:kafka:us-east-1:123456789012:cluster/msk-cluster/xxxxx-xxxx-xxxxx` 
- Usually you have to rerun `terragrunt apply` (or `terraform apply`) to fully reconcile the resource

Note that the import part does not change the resource configuration, so it's safe to run. It just tries to reconcile the remote state with the AWS resource. You can also remove the imported resource from the state with `terraform state rm xxxxx.xxxx.xxx`.

With a management account, the process might be slightly easier than with a service account - purely because you probably have less resources in the management account.

**Reply by umm0n** [reply]
Hey @autero1 , thanks for the thoughtful response! However, I think maybe I wasn't clear enough. We certainly know how to import resources, what we need to know is how to import an account itself. What steps should we take to incorporate an existing AWS account into our Gruntworks IAC (ref arch deployed, btw)? Do we need to add the existing account to accounts.json? Do we need to make sure to deploy the mgmt vpc and ecs-deployer before doing anything else?

**Reply by rhoboat** [reply]
Hey @umm0n, We have a guide for how to DIY importing the root account and child accounts.

[Import existing resources from the root account into Terraform state](https://docs.gruntwork.io/guides/build-it-yourself/landing-zone/deployment-walkthrough/import-existing-resources-from-the-root-account-into-terraform-state/)

Please note:
- ℹ️ The context of this guide: It is one part of the full DIY Landing Zone account structure guide (which walks through how to deploy all the accounts).
- ➖ This guide might be outdated.
- ➖ Importing might be quite painful.
- ➕ This guide is accurate regarding what order to `apply` the security-baselines to the accounts.

**Reply by umm0n** [reply]
Thank you, I believe this is what I was looking for!
